samples/test-samples.cmake

```diff
@@ -160,8 +160,15 @@ foreach (i RANGE ${len})
     # Build with the CMake package
     message(
       STATUS "Samples test '${SAMPLE}-${CRYPTO_LIB}' with CMake running...")
-    execute_process(COMMAND ${CMAKE_COMMAND} --build ${SAMPLE_BUILD_DIR}
-                            --target run RESULT_VARIABLE TEST_RESULT)
+    if (SAMPLE MATCHES "attestation")
+      execute_process(
+        COMMAND ${CMAKE_COMMAND} -E env SGX_AESM_ADDR=1 ${CMAKE_COMMAND} --build
```

Comment:
Now there is an inconsistency between the CMake and Make builds;
the former requires setting the `SGX_AESM_ADDR` when invoking while the latter sets the variable as part of the Makefile.

Given that SGX_AESM_ADDR is an env variable, can we
- Avoid setting the variable by the build script, but updating build instruction the sample's README. Something like:
```
# Use out-of-proc with CMake
SGX_AESM_ADDR=1 cmake ....

# Use in-proc with CMake
SGX_AESM_ADDR=0 cmake ....

# Use out-of-proc with Make
SGX_AESM_ADDR=1 make ....

# Use in-proc with Make
SGX_AESM_ADDR=0 make ....
```

If we want to enforce out-of-proc as OE's default, we should update the `openenclaverc` instead to set the env variable.

Reply:
Based on our discussion we do not enforce it as OE's default call path, but I moved both settings to `test-samples.cmake`.

Reply:
Thanks!